angular/projects/admin/src/app/services/document.service.ts

```diff
@@ -98,8 +98,15 @@ export class DocumentService {
       .valueChanges();
   }
 
-  query(documentTypeId: string, status: string, queryOpts: DocumentTypeQueryOptions = {}): Observable<Document[]> {
-    console.log(`[DocumentService:getDocumentTypeFields] ${documentTypeId}, query=${JSON.stringify(queryOpts)}`);
+  query(
+    documentTypeId: string,
+    status: string,
+    queryOpts: DocumentTypeQueryOptions = {},
+    nextPage?: Boolean,
+    pageIndex?: number,
+    lastDocSnap?: any,
+    backDocSnap?: any
+  ): Observable<Document[]> {
```

Comment:
Try to not bring in concepts of "pagination back and forward" in a query. 

Look at the available filtering that you can don directly on the `CollectionReference` and those are the types of parameters that you are allowed to pass in.

In this case, you could extend `DocumentTypeQueryOptions` with a optional field called `startAt` that is a document snapshot. 

Remove `lastDocSnap`, `backDocSnap` and decide which one to pass in as `startAt` before calling `query()`

Reply:
Resolved on this commit https://github.com/oddbit/tanam/pull/220/commits/4abec6e1632ed1302d8f7aa4287a184718f1352b